From 71e205b45c426c5251ccf947f41943c5dc86ed0c Mon Sep 17 00:00:00 2001
From: Hao Zheng <haoz@marvell.com>
Date: Tue, 20 Aug 2019 22:22:40 -0700
Subject: [PATCH 328/767] octeontx2-af: combine LB_STAG and LB_QINQ to one LB
 ltype

commit 369c58581073495eae851f4813a5133cc37ef92f from
git@git.assembla.com:cavium/WindRiver.linux.git

NPC parser changes to combine NPC_LT_LB_STAG and NPC_LT_LB_QINQ
to NPC_LT_LB_STAG_QINQ, to facilitate formulating a single MCAM
entry to match on packets with one or stacked vlan tags.

Change-Id: I5fa7dab5c99f2ee964123b6c81f7af695d72ba40
Signed-off-by: Hao Zheng <haoz@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/14808
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/npc.h   |   3 +-
 .../marvell/octeontx2/af/npc_profile.h        | 154 +++++++++---------
 .../ethernet/marvell/octeontx2/af/rvu_npc.c   |  12 +-
 .../marvell/octeontx2/af/rvu_npc_fs.c         |   8 +-
 4 files changed, 82 insertions(+), 95 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc.h b/drivers/net/ethernet/marvell/octeontx2/af/npc.h
index cc3079e83663..d58c9618d96d 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc.h
@@ -36,9 +36,8 @@ enum npc_kpu_la_ltype {
 enum npc_kpu_lb_ltype {
 	NPC_LT_LB_ETAG = 1,
 	NPC_LT_LB_CTAG,
-	NPC_LT_LB_STAG,
+	NPC_LT_LB_STAG_QINQ,
 	NPC_LT_LB_BTAG,
-	NPC_LT_LB_QINQ,
 	NPC_LT_LB_ITAG,
 	NPC_LT_LB_DSA,
 	NPC_LT_LB_DSA_VLAN,
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index 6a679d46c7f7..852a7fc9ff22 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -7477,7 +7477,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 0, 6, 2, 0,
 		NPC_S_KPU5_IP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7485,7 +7485,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		6, 0, 0, 2, 0,
 		NPC_S_KPU5_IP6, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7493,7 +7493,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_ARP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7501,7 +7501,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_RARP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7509,7 +7509,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_PTP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7517,7 +7517,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_FCOE, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7525,7 +7525,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 1, 0,
 		NPC_S_KPU4_MPLS, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7533,7 +7533,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 1, 0,
 		NPC_S_KPU4_MPLS, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7541,7 +7541,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 0, 0, 1, 0,
 		NPC_S_KPU4_NSH, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7549,7 +7549,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 0, 1,
 		NPC_S_NA, 0, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_CTAG_UNK,
 		0, 0, 0, 0,
 	},
@@ -7557,7 +7557,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 0, 0, 0,
 		NPC_S_KPU3_CTAG, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_STAG_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7565,7 +7565,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 0, 0, 0,
 		NPC_S_KPU3_STAG, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG|NPC_F_LB_L_WITH_STAG_STAG,
 		0, 0, 0, 0,
 	},
@@ -7669,7 +7669,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 0, 1,
 		NPC_S_NA, 0, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_UNK_ETYPE,
 		0, 0, 0, 0,
 	},
@@ -7677,7 +7677,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 0, 6, 2, 0,
 		NPC_S_KPU5_IP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7685,7 +7685,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		6, 0, 0, 2, 0,
 		NPC_S_KPU5_IP6, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7693,7 +7693,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_ARP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7701,7 +7701,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_RARP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7709,7 +7709,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_PTP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7717,7 +7717,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_FCOE, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7725,7 +7725,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 1, 0,
 		NPC_S_KPU4_MPLS, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7733,7 +7733,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 1, 0,
 		NPC_S_KPU4_MPLS, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7741,7 +7741,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 0, 0, 1, 0,
 		NPC_S_KPU4_NSH, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7749,7 +7749,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 0, 1,
 		NPC_S_NA, 0, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_CTAG_UNK,
 		0, 0, 0, 0,
 	},
@@ -7757,7 +7757,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 0, 0, 0,
 		NPC_S_KPU3_CTAG, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_QINQ_CTAG,
 		0, 0, 0, 0,
 	},
@@ -7765,7 +7765,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 0, 0, 0,
 		NPC_S_KPU3_QINQ, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_QINQ_QINQ,
 		0, 0, 0, 0,
 	},
@@ -7773,7 +7773,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 0, 1,
 		NPC_S_NA, 0, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_UNK_ETYPE,
 		0, 0, 0, 0,
 	},
@@ -8077,7 +8077,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 0, 6, 2, 0,
 		NPC_S_KPU5_IP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8085,7 +8085,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		6, 0, 0, 2, 0,
 		NPC_S_KPU5_IP6, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8093,7 +8093,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_ARP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8101,7 +8101,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_RARP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8109,7 +8109,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_PTP, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8117,7 +8117,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 2, 0,
 		NPC_S_KPU5_FCOE, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8125,7 +8125,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 1, 0,
 		NPC_S_KPU4_MPLS, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8133,7 +8133,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 1, 0,
 		NPC_S_KPU4_MPLS, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8141,7 +8141,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 0, 0, 1, 0,
 		NPC_S_KPU4_NSH, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8149,7 +8149,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 0, 0, 0,
 		NPC_S_KPU3_QINQ, 10, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_MORE_TAG | NPC_F_LB_L_WITH_QINQ_QINQ,
 		0, 0, 0, 0,
 	},
@@ -8157,7 +8157,7 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 0, 1,
 		NPC_S_NA, 0, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		NPC_F_LB_U_UNK_ETYPE,
 		0, 0, 0, 0,
 	},
@@ -8385,14 +8385,6 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_F_LB_U_UNK_ETYPE | NPC_F_LB_L_EXDSA,
 		0, 0, 0, 0,
 	},
-	{
-		NPC_ERRLEV_LB, NPC_EC_L2_K3,
-		0, 0, 0, 0, 1,
-		NPC_S_NA, 0, 0,
-		NPC_LID_LB, NPC_LT_NA,
-		0,
-		0, 0, 0, 0,
-	},
 };
 
 static struct npc_kpu_profile_action kpu3_action_entries[] = {
@@ -8984,7 +8976,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 0, 6, 1, 0,
 		NPC_S_KPU5_IP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8992,7 +8984,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		6, 0, 0, 1, 0,
 		NPC_S_KPU5_IP6, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9000,7 +8992,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_ARP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9008,7 +9000,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_RARP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9016,7 +9008,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_PTP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9024,7 +9016,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_FCOE, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9032,7 +9024,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9040,7 +9032,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9048,7 +9040,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 0, 0, 0, 0,
 		NPC_S_KPU4_NSH, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9056,7 +9048,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 0, 6, 1, 0,
 		NPC_S_KPU5_IP, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9064,7 +9056,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		6, 0, 0, 1, 0,
 		NPC_S_KPU5_IP6, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9072,7 +9064,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_ARP, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9080,7 +9072,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_RARP, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9088,7 +9080,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9096,7 +9088,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9104,7 +9096,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 0, 0, 0, 0,
 		NPC_S_KPU4_NSH, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9112,7 +9104,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_LB, NPC_EC_L2_K3_ETYPE_UNK,
 		0, 0, 0, 0, 1,
 		NPC_S_NA, 0, 1,
-		NPC_LID_LB, NPC_LT_LB_STAG,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9120,7 +9112,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 0, 6, 1, 0,
 		NPC_S_KPU5_IP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9128,7 +9120,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		6, 0, 0, 1, 0,
 		NPC_S_KPU5_IP6, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9136,7 +9128,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_ARP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9144,7 +9136,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_RARP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9152,7 +9144,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_PTP, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9160,7 +9152,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_FCOE, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9168,7 +9160,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9176,7 +9168,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9184,7 +9176,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 0, 0, 0, 0,
 		NPC_S_KPU4_NSH, 8, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9192,7 +9184,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 0, 6, 1, 0,
 		NPC_S_KPU5_IP, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9200,7 +9192,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		6, 0, 0, 1, 0,
 		NPC_S_KPU5_IP6, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9208,7 +9200,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_ARP, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9216,7 +9208,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_RARP, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9224,7 +9216,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_PTP, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9232,7 +9224,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		0, 0, 0, 1, 0,
 		NPC_S_KPU5_FCOE, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9240,7 +9232,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9248,7 +9240,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 6, 10, 0, 0,
 		NPC_S_KPU4_MPLS, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9256,7 +9248,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		2, 0, 0, 0, 0,
 		NPC_S_KPU4_NSH, 4, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
@@ -9264,7 +9256,7 @@ static struct npc_kpu_profile_action kpu3_action_entries[] = {
 		NPC_ERRLEV_LB, NPC_EC_L2_K3_ETYPE_UNK,
 		0, 0, 0, 0, 1,
 		NPC_S_NA, 0, 1,
-		NPC_LID_LB, NPC_LT_LB_QINQ,
+		NPC_LID_LB, NPC_LT_LB_STAG_QINQ,
 		0,
 		0, 0, 0, 0,
 	},
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
index 640f435cfe64..60c15dafbd54 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
@@ -910,13 +910,11 @@ static void npc_config_tx_ldata_extract(struct rvu *rvu, int blkaddr)
 	/* Layer B: Stacked VLAN (STAG|QinQ) */
 	/* Outer VLAN: 2 bytes, KW0[63:48] */
 	cfg = KEX_LD_CFG(0x01, 0x2, 0x1, 0x0, 0x6);
-	SET_KEX_LD(NIX_INTF_TX, NPC_LID_LB, NPC_LT_LB_STAG, 0, cfg);
-	SET_KEX_LD(NIX_INTF_TX, NPC_LID_LB, NPC_LT_LB_QINQ, 0, cfg);
+	SET_KEX_LD(NIX_INTF_TX, NPC_LID_LB, NPC_LT_LB_STAG_QINQ, 0, cfg);
 
 	/* Outer VLAN: 2 Bytes, KW1[15:0] */
 	cfg = KEX_LD_CFG(0x01, 0x8, 0x1, 0x0, 0x8);
-	SET_KEX_LD(NIX_INTF_TX, NPC_LID_LB, NPC_LT_LB_STAG, 1, cfg);
-	SET_KEX_LD(NIX_INTF_TX, NPC_LID_LB, NPC_LT_LB_QINQ, 1, cfg);
+	SET_KEX_LD(NIX_INTF_TX, NPC_LID_LB, NPC_LT_LB_STAG_QINQ, 1, cfg);
 
 	/* DMAC: 6 bytes, KW1[63:16] */
 	cfg = KEX_LD_CFG(0x05, 0x8, 0x1, 0x0, 0xa);
@@ -969,13 +967,11 @@ static void npc_config_rx_ldata_extract(struct rvu *rvu, int blkaddr)
 	/* Layer B: Stacked VLAN (STAG|QinQ) */
 	/* Outer VLAN: 2 bytes, KW0[63:48] */
 	cfg = KEX_LD_CFG(0x01, 0x2, 0x1, 0x0, 0x6);
-	SET_KEX_LD(NIX_INTF_RX, NPC_LID_LB, NPC_LT_LB_STAG, 0, cfg);
-	SET_KEX_LD(NIX_INTF_RX, NPC_LID_LB, NPC_LT_LB_QINQ, 0, cfg);
+	SET_KEX_LD(NIX_INTF_RX, NPC_LID_LB, NPC_LT_LB_STAG_QINQ, 0, cfg);
 
 	/* Ethertype: 2 bytes, KW0[47:32] */
 	cfg = KEX_LD_CFG(0x01, 0x8, 0x1, 0x0, 0x4);
-	SET_KEX_LD(NIX_INTF_RX, NPC_LID_LB, NPC_LT_LB_STAG, 1, cfg);
-	SET_KEX_LD(NIX_INTF_RX, NPC_LID_LB, NPC_LT_LB_QINQ, 1, cfg);
+	SET_KEX_LD(NIX_INTF_RX, NPC_LID_LB, NPC_LT_LB_STAG_QINQ, 1, cfg);
 
 	/* Layer C: IPv4 */
 	/* SIP+DIP: 8 bytes, KW2[63:0] */
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
index edbae9a7969a..ce0eab703e54 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
@@ -427,9 +427,9 @@ do {									       \
 	NPC_SCAN_HDR(NPC_DPORT_TCP, NPC_LID_LD, NPC_LT_LD_TCP, 2, 2);
 	NPC_SCAN_HDR(NPC_ETYPE_ETHER, NPC_LID_LA, NPC_LT_LA_ETHER, 12, 2);
 	NPC_SCAN_HDR(NPC_ETYPE_TAG1, NPC_LID_LB, NPC_LT_LB_CTAG, 4, 2);
-	NPC_SCAN_HDR(NPC_ETYPE_TAG2, NPC_LID_LB, NPC_LT_LB_STAG, 8, 2);
+	NPC_SCAN_HDR(NPC_ETYPE_TAG2, NPC_LID_LB, NPC_LT_LB_STAG_QINQ, 8, 2);
 	NPC_SCAN_HDR(NPC_VLAN_TAG1, NPC_LID_LB, NPC_LT_LB_CTAG, 2, 2);
-	NPC_SCAN_HDR(NPC_VLAN_TAG2, NPC_LID_LB, NPC_LT_LB_STAG, 2, 2);
+	NPC_SCAN_HDR(NPC_VLAN_TAG2, NPC_LID_LB, NPC_LT_LB_STAG_QINQ, 2, 2);
 	NPC_SCAN_HDR(NPC_DMAC, NPC_LID_LA, la_ltype, la_start, 6);
 	NPC_SCAN_HDR(NPC_SMAC, NPC_LID_LA, la_ltype, la_start, 6);
 	/* PF_FUNC is 2 bytes at 0th byte of NPC_LT_LA_IH_NIX_ETHER */
@@ -696,8 +696,8 @@ do {									      \
 				 0, ~0ULL, 0, intf);
 	if (features & BIT_ULL(NPC_OUTER_VID))
 		npc_update_entry(rvu, NPC_LB, entry,
-				 NPC_LT_LB_STAG | NPC_LT_LB_CTAG, 0,
-				 NPC_LT_LB_STAG & NPC_LT_LB_CTAG, 0, intf);
+				 NPC_LT_LB_STAG_QINQ | NPC_LT_LB_CTAG, 0,
+				 NPC_LT_LB_STAG_QINQ & NPC_LT_LB_CTAG, 0, intf);
 
 	NPC_WRITE_FLOW(NPC_DMAC, dmac, dmac_val, 0, dmac_mask, 0);
 	NPC_WRITE_FLOW(NPC_SMAC, smac, smac_val, 0, smac_mask, 0);
-- 
2.31.1

